ownCloud/Import/ImportFilesController.swift

```diff
@@ -0,0 +1,154 @@
+//
+//  ImportFilesController.swift
+//  ownCloud
+//
+//  Created by Matthias Hühne on 10.07.19.
+//  Copyright © 2019 ownCloud GmbH. All rights reserved.
+//
+
+/*
+* Copyright (C) 2019, ownCloud GmbH.
+*
+* This code is covered by the GNU Public License Version 3.
+*
+* For distribution utilizing Apple mechanisms please see https://owncloud.org/contribute/iOS-license-exception/
+* You should have received a copy of this license along with this program. If not, see <http://www.gnu.org/licenses/gpl-3.0.en.html>.
+*
+*/
+
+import UIKit
+import ownCloudSDK
+
+class ImportFilesController: NSObject {
+
+	// MARK: - Instance variables
+	var url: URL
+
+	// MARK: - Init & Deinit
+	init(url: URL) {
+		self.url = url
+	}
+
+}
+
+extension ImportFilesController {
+
+	func accountOrImportUI() -> Bool {
+		let bookmarks : [OCBookmark] = OCBookmarkManager.shared.bookmarks as [OCBookmark]
+
+		if bookmarks.count > 1 {
+			let moreViewController = self.cardViewController(for: url)
+			if let wd = UIApplication.shared.delegate?.window {
+				let vc = wd!.rootViewController
+				if let navCon = vc as? UINavigationController, let topVC = navCon.visibleViewController {
+					OnMainThread {
+						topVC.present(asCard: moreViewController, animated: true)
+					}
+				} else {
+					OnMainThread {
+						vc?.present(asCard: moreViewController, animated: true)
+					}
+				}
+
+				return true
+			}
+		} else if bookmarks.count == 1, let bookmark = bookmarks.first {
+			self.importItemWithDirectoryPicker(with: url, into: bookmark)
+
+			return true
+		}
+
+		return false
+	}
+
+	func importItemWithDirectoryPicker(with url : URL, into bookmark: OCBookmark) {
+		OCCoreManager.shared.requestCore(for: bookmark, setup: { (_, _) in
+		}, completionHandler: { (core, error) in
+			if error == nil {
+				OnMainThread {
+					let directoryPickerViewController = ClientDirectoryPickerViewController(core: core!, path: "/", selectButtonTitle: "Save here".localized, avoidConflictsWith: [], choiceHandler: { [weak self] (selectedDirectory) in
+
+						if let targetDirectory = selectedDirectory {
+							self?.importFile(url: url, to: targetDirectory, core: core)
```

Comment:
`importFile` is never called here because there's `[weak self]` for this block a few lines above and the `ImportFilesController` is released before this block is run (testing with a single account setup, YMMV).

Please make `self` non-weak here.